src/cmd/statsrelay.rs

```diff
@@ -67,60 +148,7 @@ fn main() -> anyhow::Result<()> {
     let scope = collector.scope("statsrelay");
 
     runtime.block_on(async move {
-        let backends = backends::Backends::new(scope.scope("backends"));
-
-        let backend_reloads = scope.counter("backend_reloads").unwrap();
-
-        let (sender, tripwire) = Tripwire::new();
-        let run = statsd_server::run(scope.scope("statsd_server"), tripwire, config.statsd.bind.clone(), backends.clone());
-
-        // Trap ctrl+c and sigterm messages and perform a clean shutdown
-        let mut sigint = signal(SignalKind::interrupt()).unwrap();
-        let mut sigterm = signal(SignalKind::terminate()).unwrap();
-        tokio::spawn(async move {
-            select! {
-            _ = sigint.recv() => info!("received sigint"),
-            _ = sigterm.recv() => info!("received sigterm"),
-            }
-            sender.cancel();
-        });
-
-        // Trap sighup to support manual file reloading
-        let mut sighup = signal(SignalKind::hangup()).unwrap();
-        // This task is designed to asynchronously build backend configurations,
-        // which may in turn come from other data sources or discovery sources.
-        // This inherently races with bringing up servers, to the point where a
-        // server may not have any backends to dispatch to yet, if discovery is
-        // very slow. This is the intended state, as configuration of processors
-        // and any buffers should have already been performed.
-        //
-        // SIGHUP will attempt to reload backend configurations as well as any
-        // discovery changes.
-        tokio::spawn(async move {
-            let dconfig = config.discovery.unwrap_or_default();
-            let discovery_cache = discovery::Cache::new();
-            let mut discovery_stream =
-                discovery::reflector(discovery_cache.clone(), discovery::as_stream(&dconfig));
-            loop {
-                info!("loading configuration and updating backends");
-                backend_reloads.inc();
-                let config = load_backend_configs(&discovery_cache, &backends, opts.config.as_ref()).await.unwrap();
-                let dconfig = config.discovery.unwrap_or_default();
-
-                tokio::select! {
-                    _ = sighup.recv() => {
-                        info!("received sighup");
-                        discovery_stream = discovery::reflector(discovery_cache.clone(), discovery::as_stream(&dconfig));
-                        info!("reloaded discovery stream");
-                    }
-                    Some(event) = discovery_stream.next() => {
-                        info!("updating discovery for map {}", event.0);
-                    }
-                };
-            }
-        });
-
-        run.await;
+        server(scope, config, opts).await;
```

Comment:
I think this can probably be shortened to `runtime.block_on(server(scope, config, opts))`

Reply:
+1